addons/website_event/controllers/main.py

```diff
@@ -243,11 +253,36 @@ def _process_tickets_form(self, event, form_details):
             'quantity': count,
         } for tid, count in ticket_order.items() if count]
 
+    @http.route(['/event/<model("event.event"):event>/registration/tickets'], type='jsonrpc', auth="public", methods=['POST'], website=True)
+    def registration_tickets(self, event, **post):
+        """ After slot selection, render ticket selection modal.
+        To restrict the selectable number of tickets, give the slot seats available and
+        each slot tickets seats available to the template.
+        """
+        slot_id = int(post.get('event_slot_id'))
+        slot = request.env['event.slot'].sudo().browse(slot_id)
+        slot_tickets = [
+            (slot, ticket)
+            for ticket in event.event_ticket_ids
+        ]
+        return request.env['ir.ui.view']._render_template("website_event.modal_ticket_registration", {
+            'event': event,
+            'event_slot_id': slot_id,
+            'seats_available_slot': slot.seats_available,
+            'seats_available_slot_tickets': {
+                ticket.id: availability
+                for (_, ticket), availability in zip(slot_tickets, event._get_seats_availability(slot_tickets))
+            }
+        })
+
     @http.route(['/event/<model("event.event"):event>/registration/new'], type='jsonrpc', auth="public", methods=['POST'], website=True)
     def registration_new(self, event, **post):
         tickets = self._process_tickets_form(event, post)
+        slot_id = post.get('event_slot_id', False)
+        # If slot => availability check already performed after slot selection to restrict nb of tickets.
+        # No slot => perform availability check.
```

Comment:
Isn't the availability check also performed for tickets though? Cfr the big computation part in the template?